blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java

```diff
@@ -342,6 +346,9 @@ public MetricData getDatapointsForRange(Locator locator, Range range, Granularit
             if (rollupType == null) {
                 rollupType = RollupType.BF_BASIC;
             }
+            if (type == null && rollupType == RollupType.ENUM){
+                return getEnumMetricDataForRange(locator, range, gran);
```

Comment:
Is there ever a time when "rollupType" is ENUM and you don't want to call getEnumMetricDataForRange()?  If not, why are you testing "type" for null?